src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -80,18 +81,80 @@ class ActivityFeed extends React.Component<Props, State> {
     approvalCommentFormSubmitHandler = (): void => this.setState({ isInputOpen: false });
 
     /**
-     * Creates a comment
+     * Add a placeholder pending feed item.
      *
-     * @param {string} text - Comment text
-     * @param {boolean} hasMention - If this comment contains an at mention
+     * @param {Object} itemBase - Base properties for item to be added to the feed as pending.
+     * @return {void}
+     */
+    addPendingItem = (itemBase: Object): void => {
+        const { currentUser } = this.props;
+        // create a placeholder pending feed item
+        const date = new Date().toISOString();
+        const feedItem = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...itemBase
+        };
+
+        const { feedItems } = this.state;
+        feedItems.unshift(feedItem);
```

Comment:
Typically in react it is an anti-pattern to modify state directly, and the items should be treated as immutable. You should always return a new array or object instead e.g.
```javascript
this.setState({ feedItems: [feedItem, ...this.state.feedItems] });
```

Reply:
Cleaned this up! Thanks for the tip